virttest/virt_vm.py

```diff
@@ -638,6 +638,7 @@ def get_address(self, index=0):
         :raise VMAddressVerificationError: If the MAC-IP address mapping cannot
                 be verified (using arping)
         """
+
         nic = self.virtnet[index]
         self.ip_version = self.params.get("ip_version", "ipv4").lower()
```

Comment:
The original idea is not add a new param in function define, but get it from test params dict, just like the 'ip_version':
```
self.flexible_index = "yes" == self.params.get("flexible_index", "no")
```
then no need update these callers.
And maybe an info message about using  flexible index instead of index=0 is necessary